python/thunder/rdds/images.py

```diff
@@ -436,6 +436,41 @@ def medianFilter(self, size=2):
 
         return self.filter(func=lambda x: median_filter(x, size))
 
+    def localCorr(self, neighborhood=2):
+        """
+        Correlate every pixel to the average of its local neighborhood.
+
+        This algorithm computes, for every spatial record, the correlation coefficient
+        between that record's series, and the average series of all records within
+        a local neighborhood with a size defined by the neighborhood parameter.
+        For data with three spatial keys, only neighborhoods in x and y
+        currently supported.
+
+        parameters
+        ----------
+        neighborhood: int, optional, default=2
```

Comment:
Handling sequences will be tricky here. Currently, if you pass a sequence of two numbers to the filtering operation for 3D data, it will fail because it assumes one number per dimension. The same thing will happen here if neighborhood is a list of two numbers. Ultimately, we should modify the filtering methods to handle as a special case a 2-number sequence for 3D data, and then do the same here, but for now I would say just force the neighborhood to be a scalar.